### PR TITLE
[feat]: Added logic to handle function call expression

### PR DESCRIPTION
**Pull request by mutahirtintash**

### CODE REVIEW

File: src/clarity-frontend/clarity_convert.cpp [hunk 1]
```diff
@@ -3243,15 +3243,33 @@ bool clarity_convertert::get_expr(
...
+    const nlohmann::json &callee_expr_json = expr;
 
     // 0. check if it's a clarity built-in function
+    // in any case we first need to get the arguments type 
+    // passed to the function
+    exprt first_arg;
+    nlohmann::json expression_type;
+    if (expr.contains("args"))
+    {
+      if (get_expr(expr["args"][0], nullptr, first_arg, expression_type))
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:3258]: can we replace this with claritygrammar:get_expression_args()[0] ?

File: src/clarity-frontend/clarity_convert.cpp [hunk 2]
```diff
@@ -3261,20 +3279,27 @@ bool clarity_convertert::get_expr(
 
       // populate params
       // the number of arguments defined in the template
-      size_t define_size = to_code_type(new_expr.type()).arguments().size();
+      
       // the number of arguments actually inside the json file
-      const size_t arg_size = expr["arguments"].size();
-      if (define_size >= arg_size)
+      const size_t arg_size = expr["args"].size();
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:3284]: maybe getting args at the top of this case block using get_expression_args and using those in the code would make it cleaner.

File: src/clarity-frontend/clarity_convert.cpp [hunk 3]
```diff
@@ -3283,178 +3308,51 @@ bool clarity_convertert::get_expr(
     }
 
     // 1. Get callee expr
+    // ml- change the contract.function name from solidity to 
+    //     simple function search
     if (
-      callee_expr_json.contains("nodeType") &&
-      callee_expr_json["nodeType"] == "MemberAccess")
+      callee_expr_json.contains("type") &&
+      callee_expr_json["type"] == "user_function")
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:3315]: get_expression_type

File: src/clarity-frontend/clarity_convert.cpp [hunk 4]
```diff
@@ -3283,178 +3308,51 @@ bool clarity_convertert::get_expr(
...
-      // __ESBMC_assume only handle one param.
-      if (
-        callee_expr.type().get("#clar_name").as_string().find("require") !=
-        std::string::npos)
-        break;
-    }
-    log_debug("clarity", "  @@ num_args={}", num_args);
-
-    new_expr = call;
+    log_error("ClarityGrammar::ExpressionT::CallExprClass The function is neither user nor intrinsic");
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:3351]: Slight rewording of this string please
" .... neither user defined nor an intrinsic function."

File: src/clarity-frontend/clarity_convert.cpp [hunk 5]
```diff
@@ -4514,19 +4412,21 @@ bool clarity_convertert::get_func_decl_ref(
 {
   // Function to configure new_expr that has a +ve referenced id, referring to a function declaration
   // This allow to get func symbol before we add it to the symbol table
-  assert(decl["nodeType"] == "FunctionDefinition");
+  assert(
+    decl["type"] == "user_function" ||
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:4416]: all keys of a json element can be accessed using helpers. please use those.

File: src/clarity-frontend/clarity_convert.cpp [hunk 6]
```diff
@@ -4601,72 +4501,62 @@ bool clarity_convertert::get_esbmc_builtin_ref(
...
+  if (expr["type"].get<std::string>() == "native_function")
   {
-    //  e.g. gasleft() <=> c:@gasleft
-    if (expr["expression"]["nodeType"].get<std::string>() != "Identifier")
-      // this means it's not a builtin funciton
-      return true;
+    // //  e.g. gasleft() <=> c:@gasleft
+    // if (expr["expression"]["nodeType"].get<std::string>() != "Identifier")
+    //   // this means it's not a builtin funciton
+    //   return true;
```
Comment (mali-tintash) [inline src/clarity-frontend/clarity_convert.cpp:4531]: we can remove this comment

**Review by mali-tintash — CHANGES_REQUESTED** [review]

File: src/clarity-frontend/clarity_convert.cpp (hunk 2 above)
Comment (mutahirtintash) [reply src/clarity-frontend/clarity_convert.cpp:3284]: Done

**Pull request merged**